python/cudf/cudf/_lib/csv.pyx

```diff
@@ -113,6 +117,8 @@ cdef csv_reader_options make_csv_reader_options(
     object prefix,
     object index_col,
 ) except *:
+    if isinstance(datasource, NativeFile):
+        datasource = NativeFileDatasource(datasource)
```

Comment:
Can we move this into the `make_source_info` utility that is called on the next line? It seems that both csv and parquet hit that same function to set up the source information, so doing something like the following in *that* function should take care of both formats for us:

```python
...
    elif isinstance(src[0], (Datasource, NativeFile)):
        csrc  = NativeFileDatasource(src[0]) if isinstance(src[0], NativeFile) else src[0]
        return source_info(csrc.get_datasource())
...
```